### PR TITLE
chore(migration): CRA to Vite, React 18 and other dependencies upgraded

### PR DESCRIPTION
**Pull request by ajithsimons**
## Description

- Scaffolder: CRA to Vite
- React: 17 to 18
- Test: Jest to Vitest
- Upgraded to latest catena-x shared components 
- Type updates in few places to comply with latest typescript and eslint

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by sachinargade123 — APPROVED** [review]
LGTM

**Review by adityagajbhiye9 — APPROVED** [review]
LGTM

**Comment by almadigabor** [comment]
I ran the dash tool and got the following output:

```
[main] INFO Querying Eclipse Foundation for license data for 500 items.
[main] INFO Found 160 items.
[main] INFO Querying Eclipse Foundation for license data for 188 items.
[main] INFO Found 55 items.
[main] INFO Querying ClearlyDefined for license data for 470 items.
[main] INFO Found 442 items.
[main] INFO License information could not be automatically verified for the following content:
[main] INFO 
[main] INFO npm/npmjs/-/jsdom/24.0.0
[main] INFO npm/npmjs/-/vite-plugin-checker/0.6.4
[main] INFO npm/npmjs/@rollup/rollup-android-arm-eabi/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-android-arm64/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-darwin-arm64/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-arm-gnueabihf/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-arm64-gnu/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-arm64-musl/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-powerpc64le-gnu/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-riscv64-gnu/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-s390x-gnu/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-linux-x64-gnu/4.14.1
[main] INFO npm/npmjs/@rollup/rollup-win32-arm64-msvc/4.14.1
[main] INFO npm/npmjs/@vitest/utils/1.5.0
[main] INFO 
[main] INFO This content is either not correctly mapped by the system, or requires review.
[main] INFO A review is required for npm/npmjs/@rollup/rollup-android-arm-eabi/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14395 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-arm64-musl/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14396 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-riscv64-gnu/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14397 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-arm-gnueabihf/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14398 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-android-arm64/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14399 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-x64-gnu/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14400 .
[main] INFO A review is required for npm/npmjs/-/vite-plugin-checker/0.6.4.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14401 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-s390x-gnu/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14402 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-powerpc64le-gnu/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14403 .
[main] INFO A review is required for npm/npmjs/-/jsdom/24.0.0.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14404 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-win32-arm64-msvc/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14405 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-darwin-arm64/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14406 .
[main] INFO A review is required for npm/npmjs/@rollup/rollup-linux-arm64-gnu/4.14.1.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14407 .
[main] INFO A review is required for npm/npmjs/@vitest/utils/1.5.0.
[main] INFO A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14408 .
```

File: scripts/download/org.eclipse.dash.licenses.jar
Comment (almadigabor) [inline scripts/download/org.eclipse.dash.licenses.jar:1]: Do you actually need this file in your repo? I would not add this explicitly. You can always run the dependency check workflow as well.

**Comment by almadigabor** [comment]
Hey @ajithsimons, the ticket is closed and the dependency have been approved.

**Review by almadigabor — APPROVED** [review]

**Pull request merged**